src/components/Button.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { variant } from 'styled-system';
+
+import Box from './Box';
+import Text from './Text';
+
+const ButtonIcon = ({ icon }) => {
+  return <Box mr={2}>{icon}</Box>;
+};
+
+const StyledButton = styled.button(
+  `
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+    user-select: none;
+    height: 34px;
+    border-radius: 25px;
+    border: none;
+    padding: 0 20px;
+    border-style: solid;
+    border-width: 2px;
+  `,
+  variant({
+    variants: {
+      primary: {
+        bg: 'secondary',
+        borderColor: 'secondary',
+      },
+      secondary: {
+        borderColor: 'primary',
+      },
+    },
+  })
+);
+
+const Button = ({ children, icon, ...props }) => (
+  <StyledButton type="button" {...props}>
+    <Text color="primary">{icon && <ButtonIcon icon={icon} />}</Text>
```

Comment:
should the condition also wrap `<Text color="primary">`?

Reply:
definitely